merkle/sparse_merkle_tree.go

```diff
@@ -285,15 +304,15 @@ func newLocalSubtreeWriter(rev int64, prefix []byte, depths []int, newTX newTXFu
 	}
 	tree := subtreeWriter{
 		treeRevision: rev,
-		prefix:       prefix,
+		prefix:       append(make([]byte, 0, len(prefix)), prefix...),
```

Comment:
I'll punt on that for now, but if the copies turn out to be required I'll add it then - I've added a TODO to that effect.